test/run_test.py

```diff
@@ -922,6 +1051,33 @@ def run_test_module(test: str, test_directory: str, options) -> Optional[str]:
     return message
 
 
+def run_migrated_cpp_test(test: str):
+    print_to_stderr("Running {} ... [{}]".format(test, datetime.now()))
+    command = ""
+    if test in DISTRIBUTED_TESTS:
+        if "cuda" not in BUILD_ENVIRONMENT:
+            print_to_stderr("Skipping {} ... CUDA is not enabled".format(test))
+            return None
+
+        test_reports_dir = "test/test-reports/cpp-distributed/test_distributed"
+        shell(("mkdir -p " + test_reports_dir).split())
```

Comment:
This is very prone to errors (especially if test_reports_dir contains spaces)
```suggestion
        os.makedirs(test_reports_dir, exists_ok=True)
```